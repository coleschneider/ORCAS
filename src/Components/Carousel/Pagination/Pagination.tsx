import * as React from 'react';

interface PaginationProps {
  onChangeIndex: (index: number) => void;
  index: number;
  total: number;
}

class Pagination extends React.Component<PaginationProps> {
  constructor(props) {
    super(props);

    this.handleChangeIndex = this.handleChangeIndex.bind(this);
    this.renderDot = this.renderDot.bind(this);
  }

  handleChangeIndex(index) {
    this.props.onChangeIndex(index);
  }

  renderDot(index) {
    const isActive = index === this.props.index;

    return (
      <li
        className={isActive ? 'slick-active' : undefined}
        onClick={() => this.handleChangeIndex(index)}
        id={`slick-slide1${index}`}
        key={index}
      >
        <button>{index}</button>
      </li>
    );
  }
  render() {
    const children = [];
    const { total } = this.props;
    for (let index = 0; index < total; index += 1) {
      children.push(this.renderDot(index));
    }

    return <ul className="slick-dots">{children}</ul>;
  }
}
export default Pagination;
