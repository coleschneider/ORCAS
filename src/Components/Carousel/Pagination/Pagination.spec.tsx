import * as React from 'react';
import Pagination from './Pagination';

describe('Pagination', () => {
  let wrapper;
  const pageProps = {
    total: 3,
    index: 0,
    onChangeIndex: jest.fn(index => {
      return index + 1;
    }),
  };
  const setupPage = (props = {}) => {
    wrapper = shallow(<Pagination {...pageProps} {...props} />);
  };

  it('matches the snapshot', () => {
    setupPage();
    expect(wrapper).toMatchSnapshot();
  });
  it('increments on click', () => {
    setupPage();
    wrapper
      .find('ul')
      .childAt(pageProps.index)
      .props()
      .onClick();
    expect(pageProps.onChangeIndex.mock.calls.length).toBe(1);
    expect(pageProps.onChangeIndex.mock.results[0].value).toBe(1);
  });
});
