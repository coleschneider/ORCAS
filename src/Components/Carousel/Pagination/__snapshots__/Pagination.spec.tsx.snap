// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches the snapshot 1`] = `
<ul
  className="slick-dots"
>
  <li
    className="slick-active"
    id="slick-slide10"
    key="0"
    onClick={[Function]}
  >
    <button>
      0
    </button>
  </li>
  <li
    id="slick-slide11"
    key="1"
    onClick={[Function]}
  >
    <button>
      1
    </button>
  </li>
  <li
    id="slick-slide12"
    key="2"
    onClick={[Function]}
  >
    <button>
      2
    </button>
  </li>
</ul>
`;
